esrally/reporter.py

```diff
@@ -864,15 +863,18 @@ def _line(self, metric, baseline, contender, task, unit, treat_increase_as_impro
                 formatter(contender),
                 self._diff(baseline, contender, treat_increase_as_improvement, formatter),
                 unit,
+                self._diff(baseline, contender, treat_increase_as_improvement, formatter, as_percentage=True),
             ]
         else:
             return []
 
-    def _diff(self, baseline, contender, treat_increase_as_improvement, formatter=lambda x: x):
+    def _diff(self, baseline, contender, treat_increase_as_improvement, formatter=lambda x: x, as_percentage=False):
```

Comment:
For what it's worth, on my first review, I hesitated to mention that adding a boolean parameter to a function is usually considered to be a [code smell](https://martinfowler.com/bliki/FlagArgument.html). But now that Daniel suggested a refactoring, I'm less shy. :) https://adamj.eu/tech/2021/07/10/python-type-hints-how-to-avoid-the-boolean-trap/ explains why that's a problem and what the alternatives are in modern Python.

(It's fine in this case as it's a small private function only called twice: I'm not asking to change your code. But maybe you'll find those references useful.)

Reply:
Thank you for the references! They're great. It's funny because my first (uncommitted) approach was actually to implement this as a seperate private function called `diff_percentage`, but I was worried that there was too much code duplication.